File: 1614.py
class Solution(object):
    def maxDepth(self, s):
        """
        :type s: str
        :rtype: int
        """
        if s == "" or ("(" not in s):
            return 0
        
        depth = 0
        answer = 0
        
        for i in s:
            if i == "(":
                depth += 1
            if i == ")":
                depth -= 1
            if depth > answer:
                answer = depth
        return answer

File: 1614.cpp
class Solution {
public:
    int maxDepth(string s) {
        int cur = 0, max = 0;
        for(auto c: s){
            if(c == '('){
                cur++;
                if(cur > max){
                    max = cur;
                }
            }
            else if(c == ')'){
                cur--;
            }
        }
        return max;
    }
};